src/plus/integrations/providers/azure/azure.ts

```diff
@@ -65,6 +72,61 @@ export class AzureDevOpsApi implements Disposable {
 		this._workItemStates.clear();
 	}
 
+	@debug<AzureDevOpsApi['getPullRequestForBranch']>({ args: { 0: p => p.name, 1: '<token>' } })
+	public async getPullRequestForBranch(
+		provider: Provider,
+		token: string,
+		owner: string,
+		repo: string,
+		branch: string,
+		options: {
+			baseUrl: string;
+		},
+	): Promise<PullRequest | undefined> {
+		const scope = getLogScope();
+		const [projectName, _, repoName] = repo.split('/');
+
+		try {
+			const prResult = await this.request<{ value: AzurePullRequest[] }>(
+				provider,
+				token,
+				options?.baseUrl,
+				`${owner}/${projectName}/_apis/git/repositories/${repoName}/pullRequests`,
+				{
+					method: 'GET',
+				},
+				scope,
+			);
+
+			const pr = prResult?.value.find(pr => pr.sourceRefName.endsWith(branch));
+			if (pr == null) return undefined;
+
+			return new PullRequest(
+				provider,
+				{
+					id: pr.createdBy.id,
+					name: pr.createdBy.displayName,
+					avatarUrl: pr.createdBy.imageUrl,
+					url: pr.createdBy.url,
+				},
+				pr.pullRequestId.toString(),
+				pr.pullRequestId.toString(),
+				pr.title,
+				getPullRequestUrl(options.baseUrl, owner, projectName, repoName, pr.pullRequestId),
+				{
+					owner: owner,
+					repo: repo,
+				},
+				azurePullRequestStatusToState(pr.status),
+				new Date(pr.creationDate),
+				new Date(pr.creationDate),
+			);
```

Comment:
In the near future it would be nice to abstract this part into a `fromAzurePullRequest` similar to our `fromGitHubPullRequest` function.